File: backend/app/schemas/badge.py
from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime

from app.core.enum import BadgeStatus

# ==========+++++==========+++++==========
class CreateBadgeRequest(BaseModel):
    badge_code: str = Field(..., min_length=2, max_length=10)
    badge_status: BadgeStatus = BadgeStatus.AVAILABLE

# ==========+++++==========+++++==========
class BadgeResponse(BaseModel):
    badge_id: int
    badge_code: str
    badge_status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/badge.ts
export type BadgeStatus = "available" | "in_use" | "lost" | "disabled";

export interface Badge {
    badge_id: number;
    badge_code: string;
    badge_status: BadgeStatus;
    created_at: string;
}

export interface CreateBadge {
    badge_code: string;
}
